linux/Platform.c

```diff
@@ -681,6 +695,115 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
    return true;
 }
 
+void Platform_getBacktrace(Vector* frames, const BacktracePanel* panel, char** error) {
```

Comment:
It wouldn't hurt to initialize `*error` at the start of the function.
```suggestion
void Platform_getBacktrace(Vector* frames, const BacktracePanel* panel, char** error) {
   *error = NULL;

```

Reply:
What do you think about adding an assert that checks if `*error` is not null? Is it a good idea?
```c
assert(*error != NULL)
```

Reply:
The issue with output parameters is always how you handle pre-initialized values. As there is just a single caller, an `assert(error && !*error);` should be fine.

Reply:
The point is, you are using the `error` parameter for output only, not for input. It should be a good practice for the _callee_ to initialize the output value to something stable. Expecting the _caller_ to do it is more prone to errors. And notice that you only call this function in one place, that extra line of initialization shouldn't add any extra code from the compiler when it inlines it.

Reply:
@Explorer09 I agree with adding the initializer inside the function. My point with the `assert` covering both `error` and `*error` is about catching a mal-behaved caller that passes some non-NULL value for the error return value.

An alternative would be change the API for `void Platform_getBacktrace(Vector* frames, const BacktracePanel* panel, char** error);` to `char* Platform_getBacktrace(Vector* frames, const BacktracePanel* panel);` (ignoring my comment below regarding the `BacktracePanel*` for a moment).

Reply:
Another note. I believe the whole function should be wrapped in a `#ifdef HAVE_BACKTRACE` ... `#endif` conditional.

Also we can replace the `"The backtrace screen is not implemented"` dummy screen and dummy message with a compule time error (`#error "Backtrace is enabled without engine selected."`).

Reply:
> My point with the `assert` covering both `error` and `*error` is about catching a mal-behaved caller that passes some non-NULL value for the error return value.

I was looking at `asprintf(3)` that would return the output string in a similar way. If `asprintf` doesn't need an assertion for this, we won't need it either.

`assert(error)` can be covered by `ATTR_NONNULL` to the whole function. `assert(!*error)` won't do really much here, because whether `*error` is a `malloc`'d buffer should be tracked by the compiler through other means.

Reply:
> An alternative would be change the API for void `Platform_getBacktrace(Vector* frames, const BacktracePanel* panel, char** error);` to `char* Platform_getBacktrace(Vector* frames, const BacktracePanel* panel);` (ignoring my comment below regarding the `BacktracePanel*` for a moment).

I'm not a fan of this alternative. I chose `char **error` because that indicates to the developer this variable is for error. I found this pattern when I learned Objective-C and `NSError`. Returning `char *` that returns an error, I think, it's not very clear on the data returned.

> Also we can replace the "The backtrace screen is not implemented" dummy screen and dummy message with a compule time error (`#error "Backtrace is enabled without engine selected.")`.

I don't understand the compute time error. If during the configuration, the `backtrace` feature is disabled, Does it produce a compute time error? However, it's a good suggestion to change the message...

I just realized that I prefer that `error` should be an optional parameter. If the developer that uses `Plateform_getBacktrace` doesn't want to check the error, it's his choice. But it involves checking if `error` is null.